model.py

```diff
@@ -2272,18 +2273,33 @@ def author_contributors(self):
                 l = acceptable_substitutes[x.role]
                 if x.contributor not in l:
                     l.append(x.contributor)
+
+        def dedupe(l):
+            """If an item shows up multiple times in a list, 
+            keep only the first occurance.
+            """
+            seen = set()
+            deduped = []
+            for i in l:
+                if i in seen:
+                    continue
+                deduped.append(i)
+                seen.add(i)
+            return deduped
+
         if primary_author:
-            return [primary_author] + sorted(other_authors, key=lambda x: x.name)
+            return dedupe([primary_author] + sorted(other_authors, key=lambda x: x.name))
```

Comment:
Even if the same person is "supposed" to be listed as both primary author and a regular author (ie. both those facts are in the data, I'm not saying that actually makes sense), this will stop them from showing up twice in the author list we display to the patron.